### PR TITLE
Remove legacy 'CrySetFileAttributes'

### PR DESCRIPTION
**Pull request by spham-amzn**
Remove legacy function 'CrySetFileAttributes' and replace its only use with AZ::IO::SystemFile::* functions

**note** 
This replicates the original call to ```CrySetFileAttributes``` in ```CXmlNode::saveToFile``` by using AzCore functions, but does not fix the underlying flaw of this function. ```CXmlNode::saveToFile``` was relying on ```CrySetFileAttributes``` to make sure the file is writeable, and it wend directly to the OS (Windows) SetFileAttributes call, but the actual write uses the FileIOBase::Instance(), which is a layer above the final file system that can also handle aliases like ```@user```.  So the original logic will always silently fail on 'SetFileAttributes' on the xml files that it is writing to anyways because they mostly used aliases in the path, not actual paths.

Signed-off-by: Steve Pham <82231385+spham-amzn@users.noreply.github.com>

### CODE REVIEW

File: Code/Legacy/CrySystem/XML/xml.cpp [hunk 1]
```diff
@@ -1132,7 +1132,10 @@ bool CXmlNode::saveToFile(const char* fileName)
 
 bool CXmlNode::saveToFile([[maybe_unused]] const char* fileName, size_t chunkSize, AZ::IO::HandleType fileHandle)
 {
-    CrySetFileAttributes(fileName, FILE_ATTRIBUTE_NORMAL);
+    if (AZ::IO::SystemFile::Exists(fileName) && AZ::IO::SystemFile::IsWritable(fileName))
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply Code/Legacy/CrySystem/XML/xml.cpp:1135]: good catch

**Review by amznestebanpapp — APPROVED** [review]

**Comment by pollend** [comment]
I think you also want to remove CrySetFileAttributes in platform.h#339

File: Code/Legacy/CryCommon/platform_impl.cpp [hunk 2]
```diff
@@ -238,22 +237,6 @@ void InitRootDir(char szExeFileName[], uint nExeSize, char szExeRootName[], uint
     }
 }
 
-//////////////////////////////////////////////////////////////////////////
-bool CrySetFileAttributes(const char* lpFileName, uint32 dwFileAttributes)
```
Comment (pollend) [inline Code/Legacy/CryCommon/platform_impl.cpp:242]: you forgot to delete the header definition in platform.h#339

**Review by pollend — CHANGES_REQUESTED** [review]

File: Code/Legacy/CryCommon/platform_impl.cpp (hunk 2 above)
Comment (spham-amzn) [reply Code/Legacy/CryCommon/platform_impl.cpp:242]: Good catch

**Review by pollend — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**